config/routes.rb

```diff
@@ -63,20 +63,63 @@
   ##################################################
   devise_for :students, controllers: {
     sessions: 'authentication/students/sessions',
-    registrations: 'authentication/students/registrations'
+    registrations: 'authentication/students/registrations',
+    passwords: 'authentication/students/passwords'
   }
 
   devise_for :teachers, controllers: {
     sessions: 'authentication/teachers/sessions',
-    registrations: 'authentication/teachers/registrations'
+    registrations: 'authentication/teachers/registrations',
+    passwords: 'authentication/teachers/passwords'
+
   }
 
   devise_for :admins, controllers: {
     sessions: 'authentication/admins/sessions',
-    registrations: 'authentication/admins/registrations'
+    registrations: 'authentication/admins/registrations',
+    passwords: 'authentication/admins/passwords'
   }
 
   ##################################################
+  #
+  # Change Passwords
+  #
+  ##################################################
+
+  devise_scope :student do
+    post 'passwords/students/reset_request', to: 'authentication/students/passwords#send_token'
+    post 'passwords/students/reset_password', to: 'authentication/students/passwords#reset_password'
+  end
+
+  resource :student, only: [:edit] do
```

Comment:
do we need this route still? or does the :url => { :action => 'reset_password'} already allows us to run that reset_password method?